internal/generate/terraform/generate.go

```diff
@@ -1,14 +1,65 @@
 package terraform
 
 import (
+	"log"
+	"os/exec"
+	"path/filepath"
 	"sync"
 
 	"github.com/commitdev/commit0/internal/config"
 	"github.com/commitdev/commit0/internal/templator"
+	"github.com/commitdev/commit0/internal/util"
+	"github.com/kyokomi/emoji"
+	"github.com/logrusorgru/aurora"
 )
 
+// @TODO : These are specific to a k8s version. If we make the version a config option we will need to change this
+var amiLookup = map[string]string{
+	"us-east-1":    "ami-0392bafc801b7520f",
+	"us-east-2":    "ami-082bb518441d3954c",
+	"us-west-2":    "ami-05d586e6f773f6abf",
+	"eu-west-1":    "ami-059c6874350e63ca9",
+	"eu-central-1": "ami-0e21bc066a9dbabfa",
+}
+
 func Generate(t *templator.Templator, cfg *config.Commit0Config, wg *sync.WaitGroup, pathPrefix string) {
-	data := templator.GenericTemplateData{*cfg}
+	if cfg.Infrastructure.AWS.EKS.WorkerAMI == "" {
+		ami, found := amiLookup[cfg.Infrastructure.AWS.Region]
+		if !found {
+			log.Fatalln(aurora.Red(emoji.Sprintf(":exclamation: Unable to look up an AMI for the chosen region")))
+		}
+
+		cfg.Infrastructure.AWS.EKS.WorkerAMI = ami
+	}
+	data := templator.GenericTemplateData{Config: *cfg}
 
 	t.Terraform.TemplateFiles(data, false, wg, pathPrefix)
 }
+
+// Execute terrafrom init & plan
+func Execute(config *config.Commit0Config, pathPrefix string) {
+	// @TODO : Change this check. Most likely we should discover the accountid
+	if config.Infrastructure.AWS.AccountId != "" {
+		log.Println("Preparing aws environment...")
+
+		envars := util.MakeAwsEnvars(util.GetSecrets())
+
+		pathPrefix = filepath.Join(pathPrefix, "kubernetes/terraform")
+
+		// @TODO : A check here would be nice to see if this stuff exists first, mostly for testing
+		log.Println(aurora.Cyan(emoji.Sprintf(":alarm_clock: Initializing remote backend...")))
+		util.ExecuteCommand(exec.Command("terraform", "init"), filepath.Join(pathPrefix, "bootstrap/remote-state"), envars)
```

Comment:
this file has been moved to `/terraform` and not where the current `pathPrefix` is pointed towards